src/fs/filesystem.ts

```diff
@@ -138,27 +162,44 @@ export class FileSystem {
 
   async runOnTree<a>(
     path: string,
-    updateTree: boolean,
+    updateTree: boolean, // ie. do a mutation
```

Comment:
I'm actually unclear — do a mutation where? On IPFS, locally, remote IPFS?

May be clearer as `updateOnIPFS` or `update

There's a code smell here for separation of concerns. Command/query separation.

Reply:
A mutation both locally and on IPFS, and if the user has set up a `syncHook`, the DNSLink as well. Yeah might refactor this later.